runtime-transaction/src/compute_budget_instruction_details.rs

```diff
@@ -23,7 +24,12 @@ pub(crate) struct ComputeBudgetInstructionDetails {
     requested_compute_unit_price: Option<(u8, u64)>,
     requested_heap_size: Option<(u8, u32)>,
     requested_loaded_accounts_data_size_limit: Option<(u8, u32)>,
-    num_non_compute_budget_instructions: u32,
+    // instructions counts
+    num_compute_budget_instructions: u32,
+    num_builtin_instructions: u32,
+    num_non_builtin_instructions: u32,
```

Comment:
these can be u16

Reply:
It could, made them `u32` mainly because they are used to calculate cu-limits, which is `u32`, saves a bunch of `try_into()`. Honestly I don't see it, but is changing them into `u16` necessary?

Reply:
u16 -> u32 is not a `try_into` you can just do `u32::from` (it can't fail since there are more bytes available).

It saves some space per transaction, which seems reasonable to me. It's not that much relative to other parts, but ~0.5MiB for our working set of txs in banking stage to save these 6 bytes/tx.

Reply:
Instead of storing all of the instruction counts here, can we compute the default compute unit limit for before/after the `reserve_minimal_cus_for_builtin_instructions` feature gate and then inside `sanitize_and_convert_to_compute_budget_limits` use the appropriate default depending on the feature gate status?